### PR TITLE
Made additions to quantumcode.py

### PR DESCRIPTION
**Pull request by ojha-aditya**
Made additions to quantumcode.py as was discussed in issue #98. Created a Jupyter notebook for quantumcode, imported necessary libraries and made a locally testable version.

Also created another Jupyter notebook quantumcode_IBM.ipynb that works on IBM Quantum computers.

### CODE REVIEW

**Comment by SchrodingersStruggle** [comment]
I think this file should most likely be edited to be runnable on IBM quantum computers. If you are only simulating locally on it, this is essentially the same code as the file in the Jupyter notebook section of the repository (#56). It isn't that difficult to write that up, you'll just need to add a backend, add the section from step 2 of the website you linked, and use the IBM estimator rather than the state vector estimator. All of that can be found on IBM's website

**Comment by ojha-aditya** [comment]
Yes I realized that. That is why I created a separate Jupyter notebook quantumcode_IBM.ipynb just now. This one works on the IBM Quantum computers when the IBM Quantum API token is entered. Tried that on my codespace environment.

File: quantumcode_IBM.ipynb
Comment (laserlab) [inline quantumcode_IBM.ipynb:1]: empty file, please clarify

File: quantumcode.ipynb
Comment (laserlab) [inline quantumcode.ipynb:1]: Use markdown cells for documentation code
clear all cells before committing in jupyter notebooks to keep files small

**Review by laserlab — CHANGES_REQUESTED** [review]
- markdown cells
- clear outputs in jupyter notebooks
- clarify empty file

File: quantumcode_IBM.ipynb
Comment (ojha-aditya) [reply quantumcode_IBM.ipynb:1]: checked just now, somehow it does not sync the IBM compatible Jupyter notebook and gives the following error

```
> git pull --tags origin ojha-aditya/quantumcode-jupyternotebook-branch
From https://github.com/ojha-aditya/CP1-24-HW1
 * branch            ojha-aditya/quantumcode-jupyternotebook-branch -> FETCH_HEAD
hint: You have divergent branches and need to specify how to reconcile them.
hint: You can do so by running one of the following commands sometime before
hint: your next pull:
hint:
hint:   git config pull.rebase false  # merge
hint:   git config pull.rebase true   # rebase
hint:   git config pull.ff only       # fast-forward only
hint:
hint: You can replace "git config" with "git config --global" to set a default
hint: preference for all repositories. You can also pass --rebase, --no-rebase,
hint: or --ff-only on the command line to override the configured default per
hint: invocation.
fatal: Need to specify how to reconcile divergent branches.
```

Do you know what I could do to resolve this.

File: quantumcode.ipynb
Comment (ojha-aditya) [reply quantumcode.ipynb:1]: Done. Would commit as soon as the error I mentioned above can be resolved

File: quantumcode_IBM.ipynb
Comment (ojha-aditya) [reply quantumcode_IBM.ipynb:1]: I was able to do it earlier, so I can't understand what changed

File: quantumcode.ipynb
Comment (ojha-aditya) [reply quantumcode.ipynb:1]: I have actually committed but don't know why the changes only reflect in my codespace!

File: quantumcode_IBM.ipynb
Comment (laserlab) [reply quantumcode_IBM.ipynb:1]: https://stackoverflow.com/questions/71768999/how-to-merge-when-you-get-error-hint-you-have-divergent-branches-and-need-to-s

File: quantumcode.ipynb
Comment (laserlab) [reply quantumcode.ipynb:1]: you have to sync or push as well, as we discussed in class

Comment (ojha-aditya) [reply quantumcode.ipynb:1]: Done.

File: quantumcode_IBM.ipynb
Comment (ojha-aditya) [reply quantumcode_IBM.ipynb:1]: Thank you! The changes are successfully committed

**Pull request merged**